lac/block_linear_operator_05.debug fails

Bringing the discussion from #1859 into an issue of its own: lac/block_linear_operator_05.debug fails with this error message:
```
2440: /node/bangerth/trunk/dealii/tests/lac/block_linear_operator_05.cc:47:94: error: call of overloaded ‘block_diagonal_operator(<brace-enclosed initializer list>)’ is ambiguous
2440: /node/bangerth/trunk/dealii/tests/lac/block_linear_operator_05.cc:47:94: note: candidates are:
2440: /node/bangerth/trunk/dealii/include/deal.II/lac/block_linear_operator.h:563:1: note: dealii::BlockLinearOperator<Range, Domain> dealii::block_diagonal_operator(const std::array<dealii::LinearOperator<typename Range::BlockType, typename Domain::BlockType>, m>&) [with long unsigned int m = 2ul, Range = dealii::TrilinosWrappers::MPI::BlockVector, Domain = dealii::TrilinosWrappers::MPI::BlockVector, typename Domain::BlockType = dealii::TrilinosWrappers::MPI::Vector, typename Range::BlockType = dealii::TrilinosWrappers::MPI::Vector]
2440: /node/bangerth/trunk/dealii/include/deal.II/lac/block_linear_operator.h:607:1: note: dealii::BlockLinearOperator<Range, Domain> dealii::block_diagonal_operator(const dealii::LinearOperator<typename Range::BlockType, typename Domain::BlockType>&) [with long unsigned int m = 2ul, Range = dealii::TrilinosWrappers::MPI::BlockVector, Domain = dealii::TrilinosWrappers::MPI::BlockVector, typename Domain::BlockType = dealii::TrilinosWrappers::MPI::Vector, typename Range::BlockType = dealii::TrilinosWrappers::MPI::Vector]
2440: /node/bangerth/trunk/dealii/tests/lac/block_linear_operator_05.cc:47:94: error: unable to deduce ‘auto’ from ‘<expression error>’
2440: /node/bangerth/trunk/dealii/tests/lac/block_linear_operator_05.cc:48:92: error: call of overloaded ‘block_diagonal_operator(<brace-enclosed initializer list>)’ is ambiguous
2440: /node/bangerth/trunk/dealii/tests/lac/block_linear_operator_05.cc:48:92: note: candidates are:
2440: /node/bangerth/trunk/dealii/include/deal.II/lac/block_linear_operator.h:563:1: note: dealii::BlockLinearOperator<Range, Domain> dealii::block_diagonal_operator(const std::array<dealii::LinearOperator<typename Range::BlockType, typename Domain::BlockType>, m>&) [with long unsigned int m = 2ul, Range = dealii::TrilinosWrappers::BlockVector, Domain = dealii::TrilinosWrappers::BlockVector, typename Domain::BlockType = dealii::TrilinosWrappers::Vector, typename Range::BlockType = dealii::TrilinosWrappers::Vector]
2440: /node/bangerth/trunk/dealii/include/deal.II/lac/block_linear_operator.h:607:1: note: dealii::BlockLinearOperator<Range, Domain> dealii::block_diagonal_operator(const dealii::LinearOperator<typename Range::BlockType, typename Domain::BlockType>&) [with long unsigned int m = 2ul, Range = dealii::TrilinosWrappers::BlockVector, Domain = dealii::TrilinosWrappers::BlockVector, typename Domain::BlockType = dealii::TrilinosWrappers::Vector, typename Range::BlockType = dealii::TrilinosWrappers::Vector]
2440: /node/bangerth/trunk/dealii/tests/lac/block_linear_operator_05.cc:48:92: error: unable to deduce ‘auto’ from ‘<expression error>’
```

--- a/tests/lac/block_linear_operator_05.cc
+++ b/tests/lac/block_linear_operator_05.cc
@@ -44,8 +44,11 @@ int main(int argc, char *argv[])
   auto op_b = linear_operator<TrilinosWrappers::MPI::BlockVector>(b);
   auto op_b3 = linear_operator<TrilinosWrappers::BlockVector>(b);
 
-  auto op_c = block_diagonal_operator<2, TrilinosWrappers::MPI::BlockVector >({{ op_a, op_a}});
-  auto op_c2 = block_diagonal_operator<2, TrilinosWrappers::BlockVector >({{ op_a2, op_a2}});
+  typedef LinearOperator<TrilinosWrappers::MPI::Vector,TrilinosWrappers::MPI::Vector> Op_MPI;
+  typedef LinearOperator<TrilinosWrappers::Vector,TrilinosWrappers::Vector> Op;
+
+  auto op_c = block_diagonal_operator<2, TrilinosWrappers::MPI::BlockVector >(std::array<Op_MPI,2> ({{ op_a, op_a}}));
+  auto op_c2 = block_diagonal_operator<2, TrilinosWrappers::BlockVector >(std::array<Op,2> ({{ op_a2, op_a2}}));
 
   deallog << "OK" << std::endl;
 

